### PR TITLE
refactor documentation

### PR DESCRIPTION
**Pull request by SwRaw**
Refactoring the documentation to adhere to the standards

### CODE REVIEW

**Comment by SwRaw** [comment]
@lpaoletti Please review

**Comment by SwRaw** [comment]
@lancesix Please review

File: docs/conceptual/user-guide.rst [hunk 1]
```diff
@@ -0,0 +1,170 @@
+.. meta::
+   :description: A library that can be loaded by ROCr to print the AMDGPU wavefront states
+   :keywords: ROCdebug-agent, ROCm, library, tool, rocr
+
+.. _user-guide:
+
+===========================
+ROCdebug-agent user guide
```
Comment (lpaoletti) [inline docs/conceptual/user-guide.rst]:
```suggestion
ROCm Debug Agent user guide
```

Comment (lpaoletti) [reply docs/conceptual/user-guide.rst]: We should try to use the proper name as noted on the What is ROCm page. 

Fix it here, too
image

File: docs/conceptual/user-guide.rst [hunk 2]
```diff
@@ -0,0 +1,170 @@
+.. meta::
+   :description: A library that can be loaded by ROCr to print the AMDGPU wavefront states
+   :keywords: ROCdebug-agent, ROCm, library, tool, rocr
```
Comment (lpaoletti) [inline docs/conceptual/user-guide.rst]:
```suggestion
   :keywords: ROCdebug-agent, ROCm, library, tool, rocr, ROCm Debug Agent
```

File: docs/index.rst [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+.. meta::
+   :description: A library that can be loaded by ROCr to print the AMDGPU wavefront states
+   :keywords: ROCdebug-agent, ROCm, library, tool, rocr
 
-.. include:: ../README.md
-   :parser: myst_parser.sphinx_
+.. _index:
+
+==============================
+ROCdebug-agent documentation
```
Comment (lpaoletti) [inline docs/index.rst]:
```suggestion
ROCm Debug Agent documentation
```

File: docs/index.rst [hunk 4]
```diff
@@ -1,6 +1,37 @@
...
+==============================
+
+The ROCm Debug Agent (ROCdebug-agent) is a library that can be loaded by the ROCm software runtime (ROCr) to provide the following functionalities:
```
Comment (lpaoletti) [inline docs/index.rst]:
```suggestion
The ROCm Debug Agent (ROCdebug-agent) is a library that can be loaded by the [ROCm software runtime (ROCr)](LINK TO ROCR) to provide the following functionalities:
```

File: docs/index.rst [hunk 5]
```diff
@@ -1,6 +1,37 @@
...
+
+- Print the state of all AMDGPU wavefronts that cause a queue error (such as, a memory violation, executing a ``s_trap 2``, or executing an illegal instruction).
+
+- Print the state of all AMDGPU wavefronts by sending a SIGQUIT signal to the process using ``kill -s SIGQUIT <pid>`` command or by pressing ``Ctrl-\``, while the program is executing.
+
+This functionality is provided for all AMDGPUs supported by the ROCm Debugger API Library (ROCdbgapi).
```
Comment (lpaoletti) [inline docs/index.rst]:
```suggestion
This functionality is provided for all AMDGPUs supported by the [ROCm Debugger API Library (ROCdbgapi)](SHOULD BE A LINK).
```

File: docs/install/installation.rst [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+.. code-block:: shell
+
+    cd rocm-debug-agent
+    mkdir build && cd build
+    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=../install ..
+    make
+
+To specify the location for the installation, use ``CMAKE_INSTALL_PREFIX``. The default location is ``/usr``.
+
+To specify a list of paths separated by `;` that are used to locate the ``cmake`` modules, use ``CMAKE_MODULE_PATH``. It is used to locate the HIP ``cmake`` modules required to build the tests. The default is location is ``/opt/rocm/hip/cmake``.
```
Comment (lpaoletti) [inline docs/install/installation.rst]:
```suggestion
To specify a list of paths (separated by semicolons) that are used to locate the ``cmake`` modules, use ``CMAKE_MODULE_PATH``. It is used to locate the HIP ``cmake`` modules required to build the tests. The default is location is ``/opt/rocm/hip/cmake``.
```

File: docs/install/installation.rst [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    yum install gcc gcc-c++ make cmake elfutils-libelf-devel elfutils-devel
+
+  - For SLES 15 Service Pack 1:
+
+  .. code:: shell
+
+    zypper install gcc gcc-c++ make cmake libelf-devel libdw-devel
+
+- Python version 3.6 or later to run the tests.
```
Comment (lpaoletti) [inline docs/install/installation.rst]:
```suggestion
- Python 3.6 or later to run the tests.
```

File: docs/install/installation.rst [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+- ROCdbgapi library. This can be installed using the ROCdbgapi package as part of the ROCm release. See the instructions to install `ROCdbgapi library <https://rocm.docs.amd.com/projects/ROCdbgapi/en/latest/>`_.
+
+Build and install
+-------------------
+
+An example command-line to build and install the ROCdebug-agent library on Linux:
```
Comment (lpaoletti) [inline docs/install/installation.rst]:
```suggestion
An example command line to build and install the ROCdebug-agent library on Linux:
```

File: docs/install/installation.rst [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+   :description: A library that can be loaded by ROCr to print the AMDGPU wavefront states
+   :keywords: ROCdebug-agent, ROCm, library, tool, rocr
+
+.. _installation:
+
+==================================
+Installation
+==================================
+
+This document provides information required to build and install ROCdebug-agent library.
```
Comment (lpaoletti) [inline docs/install/installation.rst]:
```suggestion
This document provides information required to build and install ROCm Debug Agent (ROCdebug-agent) library.
```

File: docs/conceptual/user-guide.rst [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+.. _user-guide:
+
+===========================
+ROCdebug-agent user guide
+===========================
+
+To display the source text location with the machine code instructions around the wavefront's pc, compile the AMDGPU code objects with ``-ggdb``.  In addition, you can optionally user ``-O0`` to achieve a more intuitive display of the source text location
+as higher optimization levels can help to reorder machine code instructions. When ``-ggdb`` is not used, the source line information is not available and only machine code instructions starting at the
+wavefront's pc are printed.
```
Comment (lpaoletti) [inline docs/conceptual/user-guide.rst]:
```suggestion
To display the source text location with the machine code instructions around the wavefront's pc, compile the AMDGPU code objects with ``-ggdb``.  In addition, you can optionally use ``-O0`` to achieve a more intuitive display of the source text location, as higher optimization levels can help to reorder machine code instructions. When ``-ggdb`` is not used, the source line information is unavailable, and only machine code instructions starting at the
wavefront's pc are printed.
```

File: docs/conceptual/user-guide.rst [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+
+To use the ROCdebug-agent, set the ``HSA_TOOLS_LIB`` environment variable to the file name or path of the library and the ``HSA_ENABLE_DEBUG`` environment variable to ``1``.
+
+.. code:: shell
+
+    HSA_TOOLS_LIB=/opt/rocm/lib/librocm-debug-agent.so.2 HSA_ENABLE_DEBUG=1 ./my_program
+
+If the application encounters a triggering event, ROCdebug-agent prints the state of some or all AMDGPU wavefronts.
+
+See a sample print out:
```
Comment (lpaoletti) [inline docs/conceptual/user-guide.rst]:
```suggestion
See a sample printout:
```

File: docs/conceptual/user-guide.rst [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+        0x7fd4f100d0ec <+236>:    s_endpgm
+
+    End of disassembly.
+    Aborted (core dumped)
+
+The supported triggering events are:
+
+- **Memory fault**
+
+A memory fault occurs when an AMDGPU accesses an inaccessible page. This leads to printing information about the memory fault as shown below:
```
Comment (lpaoletti) [inline docs/conceptual/user-guide.rst]:
```suggestion
A memory fault occurs when an AMDGPU accesses an inaccessible page. This leads to printing information about the memory fault, as shown below:
```

File: docs/install/installation.rst (hunk 9 above)
Comment (SwRaw) [reply docs/install/installation.rst]: Done.

File: docs/install/installation.rst (hunk 8 above)
Comment (SwRaw) [reply docs/install/installation.rst]: Done.

File: docs/conceptual/user-guide.rst (hunk 1 above)
Comment (SwRaw) [reply docs/conceptual/user-guide.rst]: Done

**Pull request merged**